subsys/ipc/ipc_service/lib/pbuf.c

```diff
@@ -24,6 +26,20 @@ static uint32_t idx_wrap(uint32_t len, uint32_t idx)
 	return (idx >= len) ? (idx % len) : (idx);
 }
 
+static void cache_flush(void *ptr, size_t data)
+{
+	if (IS_ENABLED(CONFIG_PBUF_CACHE_HANDLING)) {
```

Comment:
This condition should be per pbuf instance, not global. If a CPU has one pbuf used for internal communication and another for IPC, only one of the instances should have cache management enabled.

On the other hand, we can target scenarios in which all pbuf are internal and then this global config makes sense. But it should be guarded by something like `depends on !IPC`.

In the systems including both internal and IPC pbufs we can by default have cache enabled for all instances. Potentially adding per-isntance cache configuration in the future to improve the performance.